src/clock.ts

```diff
@@ -0,0 +1,49 @@
+import EventEmitter from './event-emitter.js'
+
+type ClockEvents = {
+  tick: []
+}
+
+export class Clock extends EventEmitter<ClockEvents> {
+  private unsubscribe: () => void = () => undefined
+
+  private startTime = 0
+  private currentTime = 0
+  private isPaused = false
+
+  private updateTime(timestamp: number): void {
+    if (!this.isPaused) {
+      this.currentTime = timestamp - this.startTime
+      this.emit('tick')
+    }
+  }
+
+  start(): void {
+    this.startTime = performance.now()
+    this.currentTime = 0
+    this.isPaused = false
+
+    this.unsubscribe = this.on('tick', () => {
+      requestAnimationFrame(this.updateTime.bind(this))
```

Comment:
We don't really have to update the time on each tick. You could just return `performance.now() - startTime` from `getCurrentTime`. On pause, you could save the intermediate duration into a variable. This is both much more accurate and efficient.

Check out thow time is tracked in this class: https://github.com/katspaugh/wavesurfer.js/blob/main/src/webaudio.ts#L148-L150

With this in mind, we could get away with just the Timer class + tracking the recording duration in the Record plugin itself, w/o creating a new Clock class.